tree-sitter-dtd/grammar.js

```diff
@@ -330,19 +334,16 @@ module.exports = grammar({
 
     EncName: _ => /[A-Za-z][A-Za-z0-9._\-]*/,
 
-    PI: $ => prec.left(seq(
+    PI: $ => seq(
       '<?',
-      // FIXME: disallow /[Xx][Mm][Ll]/
-      alias($.Name, $.PITarget),
-      // FIXME: disallow '?>'
-      O(seq($._S, /.*/)),
+      alias($._non_xml_name, $.PITarget),
+      O(seq($._S, /(\?[^>]|[^?]*)/)),
       '?>'
-    )),
+    ),
 
-    Comment: _ => prec.left(token(
-      // FIXME: disallow '--'
-      seq('<!--', /(.|[\r\n])*/, '-->')
-    )),
+    Comment: _ => token(
+      seq('<!--', /(([^-])|(-[^-])|[\r\n])*/, '-->')
```

Comment:
This doesn't work either.

```javascript
// expected: '-' | actual: ''
/(([^-])|(-[^-])|[\r\n])*/.exec('-')[0]
```

Reply:
what about this? `choice('-', /(([^-])|(-[^-])|[\r\n])*/),`

just allowing `-` in the regex too wouldn't work, but the scanner is always an option

Reply:
That still doesn't work.